index.js

```diff
@@ -49,8 +52,10 @@ module.exports = (options = {}) => {
     if (!(ctx.compress === true || filter(ctx.response.type))) return
 
     // identity
-    const encoding = ctx.acceptsEncodings('gzip', 'deflate', 'identity')
-    if (!encoding) ctx.throw(406, 'supported encodings: gzip, deflate, identity')
+    let encoding = null
+    if (brotliSupport && brotliOptions !== null) encoding = ctx.acceptsEncodings('br') // 1st choice
```

Comment:
why is this first choice? shouldn't we honor what the client requests?

Reply:
i.e. we should be doing like...

```js
const acceptedEncodings = ['gzip', 'deflate', 'identity']
if (brotliSupport && brotliOptions !== null) acceptedEncodings.unshift('br')
const encoding = ctx.acceptsEncodings(...acceptedEncodings)
```

Reply:
this is my only concern as this is implementation is technically against specifications

Reply:
Sounds good. I will update the PR with:

1. `brotliOptions` => `brotli`
2. Rely on `acceptsEncodings()` for selecting the encoding.

Please let me know, if you change your mind on number 1 above.

A sidenote: when I was working on the patch I looked at #80 and it used a similar mechanism for other compressors: `options.gzip`, `options.deflate`. Your proposal `options.brotli` follows the same pattern. Do you want me to add a support for `options.XXX`, where `XXX` is a compressor name as defined in the standard? It could be a separate PR, if that's what you prefer.

Reply:
i guess for your sidenote, we should just stick with `br` or `brotli` in our code except for `acceptsEncodings`, right?

Reply:
Yeah, why not.